obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    elif mag_type in ['MW', 'Mw']:
+        mag = 'W'
+    elif mag_type == 'MbLg':
+        mag = 'G'
+    elif mag_type in ['Mc', 'MC']:
+        mag = 'C'
+    elif mag_type == 'M':
+        mag = 'W'  # Convert generic magnitude to moment magnitude
+        msg = ('Converting generic magnitude to being stored as moment mag')
+        warnings.warn(msg)
+    else:
+        warnings.warn(mag_type + ' is not convertible')
+        return str('')
+    return str(mag)
+
+
+def _nortoevmag(mag_type):
+    """
+    Switch from nordic type magnitude notation to obspy event magnitudes.
+
+    >>> _nortoevmag('b')
+    'mB'
+    >>> _nortoevmag('bob')
+    ''
+    """
+    mag_type = str(mag_type)
+    if mag_type == 'L':
+        mag = 'ML'
+    elif mag_type == 'b':
+        mag = 'mB'
+    elif mag_type == 's':
+        mag = 'Ms'
+    elif mag_type == 'S':
+        mag = 'MS'
+    elif mag_type == 'W':
+        mag = 'MW'
+    elif mag_type == 'G':
+        mag = 'MbLg'
+    elif mag_type == 'C':
+        mag = 'Mc'
+    else:
+        warnings.warn(mag_type + ' is not convertible')
+        return str('')
+    return str(mag)
+
+
+def readheader(sfile):
+    """
+    Read header information from a seisan nordic format S-file.
+    Returns an obspy.core.event.Catalog type: note this changed for version \
+    0.1.0 from the inbuilt class types.
+
+    :type sfile: str
+    :param sfile: Path to the s-file
+
+    :returns: :class: obspy.core.event.Event
+    """
+    with open(sfile, 'r') as f:
+        header = _readheader(f=f)
+    return header
+
+
+def _readheader(f):
+    """Internal header reader.
+    :type f: file
+    :param f: File open in read-mode.
+
+    :returns: obspy.core.event.Event"""
+    f.seek(0)
+    # Base populate to allow for empty parts of file
+    new_event = Event()
+    topline = f.readline()
+    if not len(topline.rstrip()) == 80:
+        f.close()
+        raise IOError('s-file has a corrupt header, not 80 char long')
+    f.seek(0)
+    for line in f:
+        if line[79] in [' ', '1']:
+            topline = line
+            break
+        if line[79] == '7':
+            f.close()
+            raise IOError('No header found, corrupt s-file?')
+    try:
+        sfile_seconds = int(topline[16:18])
+        if sfile_seconds == 60:
+            sfile_seconds = 0
+            add_seconds = 60
+        else:
+            add_seconds = 0
+        new_event.origins.append(Origin())
+        new_event.origins[0].time = UTCDateTime(int(topline[1:5]),
+                                                int(topline[6:8]),
+                                                int(topline[8:10]),
+                                                int(topline[11:13]),
+                                                int(topline[13:15]),
+                                                sfile_seconds,
+                                                int(topline[19:20]) *
+                                                100000)\
```

Comment:
I guess @krischer is referring to use of `UTCDateTime.strptime(date_string, format)`..